lib/matplotlib/ticker.py

```diff
@@ -1575,6 +1578,9 @@ def tick_values(self, vmin, vmax):
             subs = self._subs
 
         stride = 1
+        if not self.numticks > 1:
+            raise RuntimeError('The number of ticks must be greater than 1 '
+                               'for LogLocator.')
         while numdec / stride + 1 > self.numticks:
```

Comment:
I spent a while trying to understand this code and gave up.  The blame on it is all pretty old (2006-2008)  so I took a pass on touching it and was trying to just prevent the infinite loop.